### PR TITLE
[Cherry-pick] [Quant] [PT2] Enable batchnorm in _move_exported_model_to_eval

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Cherry-pick https://github.com/pytorch/pytorch/pull/114547 to release/2.2 branch

**Summary**
Add standalone `batchnorm` into `_move_exported_model_to_eval` to move it from `training` mode into `eval` mode.

**Test Plan**

```
python -m pytest test_mkldnn_pattern_matcher.py -k test_qat_bn_conv2d
python -u -m pytest -s -v test_quantize_pt2e.py -k test_bn_move_exported_model_to_eval
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
For densenet121 which has standalone BN can't fuse with conv:

- QAT Accuracy before this fix is: 1.696(top 1), 4.806(top 5)
- QAT Accuracy after this fix is: 72.944(top 1), 91.048 (top 5) which is close to PTQ accuracy 72.206(top 1), 90.688 (top 5)

Here is the [script](https://gist.github.com/leslie-fang-intel/c3bd82ef621a198477aeddc8393c6c89) to reproduce this issue.

**Review by andrewor14 — APPROVED** [review]

File: test/quantization/pt2e/test_quantize_pt2e_qat.py [hunk 1]
```diff
@@ -159,8 +159,8 @@ def _verify_symmetric_xnnpack_qat_numerics_helper(
         self.assertEqual(after_prepare_result_pt2e, after_prepare_result_fx)
 
         if verify_convert:
-            torch.ao.quantization.move_exported_model_to_eval(model_pt2e)
             model_pt2e = convert_pt2e(model_pt2e)
+            torch.ao.quantization.move_exported_model_to_eval(model_pt2e)
```
Comment (andrewor14) [inline test/quantization/pt2e/test_quantize_pt2e_qat.py:163]: Actually please revert this part of the change. This diff looks different from the master version?

**Review by andrewor14 — CHANGES_REQUESTED** [review]

File: test/quantization/pt2e/test_quantize_pt2e_qat.py (hunk 1 above)
Comment (leslie-fang-intel) [reply test/quantization/pt2e/test_quantize_pt2e_qat.py:163]: Sorry for the mistake. I applied the patch before our discussion. The changes in this PR should be same as https://github.com/pytorch/pytorch/pull/114547 now. Please help to take a look again.

**Comment by andrewor14** [comment]
Ok looks great, thanks @leslie-fang-intel!

**Review by andrewor14 — APPROVED** [review]
